### ISSUE TITLE
Phase 3.1: Refactor Checkout UI with Modern JavaScript Architecture

### ISSUE THREAD

**Issue by dangoriaynov** [dangoriaynov/speedy_econt_shipping#8]
## Overview
Replace the current hacky JavaScript implementation with a clean, event-driven architecture. Remove `eval()` usage, eliminate polling loops, and implement proper WooCommerce checkout integration.

## Current Problems (js.php)
1. **Security:** Uses `eval()` on line 353 to access data objects
2. **Performance:** Multiple polling loops (500ms intervals) for price detection
3. **Fragility:** Hardcoded CSS selectors, theme-dependent
4. **Maintainability:** 632 lines of inline JS with global variables
5. **UX:** Price manipulation visible to users, flickering

### File Structure
```
assets/js/
├── sesh-checkout.js          (Main checkout module)
├── sesh-location-selector.js (Region/City/Office dropdowns)
├── sesh-price-display.js     (Price formatting and display)
└── sesh-admin.js             (Admin settings page)
```

### Main Checkout Module
```javascript
// sesh-checkout.js
(function($) {
    'use strict';
    
    const SESH_Checkout = {
        // State
        state: {
            selectedCarrier: null,
            selectedDeliveryType: null, // 'office' or 'address'
            selectedLocation: {
                region: null,
                city: null,
                office: null,
            },
            isLoading: false,
        },
        
        // Configuration (passed from PHP)
        config: {},
        
        // Initialize
        init: function(config) {
            this.config = config;
            this.bindEvents();
            this.initLocationSelectors();
            this.restoreState();
        },
        
        // Event bindings
        bindEvents: function() {
            // Carrier selection
            $(document.body).on('change', 'input[name="shipping_method[0]"]', 
                this.onCarrierChange.bind(this));
            
            // Delivery type (office vs address)
            $(document.body).on('change', 'input[name="sesh_delivery_type"]',
                this.onDeliveryTypeChange.bind(this));
            
            // WooCommerce events
            $(document.body).on('updated_checkout', this.onCheckoutUpdated.bind(this));
            $(document.body).on('update_checkout', this.onCheckoutUpdating.bind(this));
        },
        
        // Carrier change handler
        onCarrierChange: function(e) {
            const carrier = this.getSelectedCarrier();
            this.state.selectedCarrier = carrier;
            
            if (this.isOurMethod(carrier)) {
                this.showDeliveryOptions(carrier);
            } else {
                this.hideDeliveryOptions();
            }
        },
        
        // When checkout updates (after AJAX)
        onCheckoutUpdated: function() {
            this.updatePriceDisplay();
            this.updateFreeShippingMessage();
        },
    };
    
    // Initialize when document ready
    $(function() {
        if (typeof sesh_checkout_config !== 'undefined') {
            SESH_Checkout.init(sesh_checkout_config);
        }
    });
    
    // Expose for debugging
    window.SESH_Checkout = SESH_Checkout;
    
})(jQuery);
```

### Location Selector Module
```javascript
// sesh-location-selector.js
const SESH_LocationSelector = {
    // Cached data
    cache: {
        regions: {},
        cities: {},
        offices: {},
    },
    
    // Initialize Select2 dropdowns
    initDropdowns: function(carrier) {
        this.initRegionDropdown(carrier);
        this.initCityDropdown(carrier);
        this.initOfficeDropdown(carrier);
    },
    
    // Region dropdown with Select2
    initRegionDropdown: function(carrier) {
        const $select = $(`#sesh_${carrier}_region`);
        
        $select.select2({
            placeholder: sesh_i18n.select_region,
            allowClear: true,
            data: this.getRegionsForCarrier(carrier),
            language: this.getSelect2Language(),
        });
        
        $select.on('change', () => this.onRegionChange(carrier));
    },
    
    // Fetch cities for region (AJAX)
    loadCities: function(carrier, regionId) {
        const cacheKey = `${carrier}_${regionId}`;
        
        if (this.cache.cities[cacheKey]) {
            return Promise.resolve(this.cache.cities[cacheKey]);
        }
        
        return $.ajax({
            url: sesh_checkout_config.ajax_url,
            method: 'POST',
            data: {
                action: 'sesh_get_cities',
                carrier: carrier,
                region_id: regionId,
                nonce: sesh_checkout_config.nonce,
            },
        }).then(response => {
            this.cache.cities[cacheKey] = response.data;
            return response.data;
        });
    },
    
    // Similar for offices...
};
```

### Price Display Module
```javascript
// sesh-price-display.js
const SESH_PriceDisplay = {
    // Update free shipping message
    updateFreeShippingMessage: function(carrier, cartTotal, threshold) {
        const $message = $('.sesh-free-shipping-message');
        
        if (threshold <= 0) {
            $message.hide();
            return;
        }
        
        const remaining = threshold - cartTotal;
        
        if (remaining <= 0) {
            $message
                .removeClass('sesh-not-free')
                .addClass('sesh-is-free')
                .html(sesh_i18n.free_shipping_eligible);
        } else {
            const formattedAmount = this.formatPrice(remaining);
            $message
                .removeClass('sesh-is-free')
                .addClass('sesh-not-free')
                .html(sesh_i18n.free_shipping_remaining.replace('%s', formattedAmount));
        }
        
        $message.show();
    },
    
    // Format price using WooCommerce settings
    formatPrice: function(amount) {
        return accounting.formatMoney(amount, {
            symbol: sesh_checkout_config.currency_symbol,
            decimal: sesh_checkout_config.decimal_separator,
            thousand: sesh_checkout_config.thousand_separator,
            precision: sesh_checkout_config.decimals,
            format: sesh_checkout_config.price_format,
        });
    },
};
```

### PHP Integration
```php
class SESH_Checkout_Scripts {
    public function enqueue_scripts() {
        if (!is_checkout()) return;
        
        // Enqueue Select2 (WooCommerce includes it)
        wp_enqueue_script('select2');
        wp_enqueue_style('select2');
        
        // Our scripts
        wp_enqueue_script(
            'sesh-checkout',
            SESH_PLUGIN_URL . 'assets/js/sesh-checkout.js',
            ['jquery', 'select2', 'accounting'],
            SESH_VERSION,
            true
        );
        
        // Localize configuration
        wp_localize_script('sesh-checkout', 'sesh_checkout_config', [
            'ajax_url' => admin_url('admin-ajax.php'),
            'nonce' => wp_create_nonce('sesh_checkout'),
            'carriers' => $this->get_enabled_carriers(),
            'currency_symbol' => get_woocommerce_currency_symbol(),
            'decimal_separator' => wc_get_price_decimal_separator(),
            'thousand_separator' => wc_get_price_thousand_separator(),
            'decimals' => wc_get_price_decimals(),
            'price_format' => get_woocommerce_price_format(),
        ]);
        
        // Translations
        wp_localize_script('sesh-checkout', 'sesh_i18n', [
            'select_region' => __('Select region...', 'speedy-econt-shipping'),
            'select_city' => __('Select city...', 'speedy-econt-shipping'),
            'select_office' => __('Select office...', 'speedy-econt-shipping'),
            'free_shipping_eligible' => __('You qualify for free shipping!', 'speedy-econt-shipping'),
            'free_shipping_remaining' => __('Add %s more for free shipping', 'speedy-econt-shipping'),
            'loading' => __('Loading...', 'speedy-econt-shipping'),
        ]);
    }
}
```

### AJAX Handlers
```php
class SESH_Ajax_Handler {
    public function __construct() {
        add_action('wp_ajax_sesh_get_cities', [$this, 'get_cities']);
        add_action('wp_ajax_nopriv_sesh_get_cities', [$this, 'get_cities']);
        add_action('wp_ajax_sesh_get_offices', [$this, 'get_offices']);
        add_action('wp_ajax_nopriv_sesh_get_offices', [$this, 'get_offices']);
    }
    
    public function get_cities() {
        check_ajax_referer('sesh_checkout', 'nonce');
        
        $carrier = sanitize_text_field($_POST['carrier']);
        $region_id = sanitize_text_field($_POST['region_id']);
        
        $cities = $this->db->get_cities($carrier, $region_id);
        
        wp_send_json_success($cities);
    }
}
```

## Requirements
- [ ] Create modular JS architecture (no inline scripts)
- [ ] Remove all `eval()` usage
- [ ] Replace polling with event-driven updates
- [ ] Use WooCommerce's `updated_checkout` event
- [ ] Implement proper AJAX handlers with nonce verification
- [ ] Add loading states for async operations
- [ ] Support keyboard navigation in dropdowns
- [ ] Ensure mobile responsiveness
- [ ] Add proper error handling with user feedback

## Acceptance Criteria
- [ ] No `eval()` in codebase
- [ ] No setInterval polling for prices
- [ ] Dropdowns load dynamically via AJAX
- [ ] Free shipping message updates without page refresh
- [ ] Works with all major WooCommerce-compatible themes
- [ ] Passes JSHint/ESLint validation
- [ ] Accessible (WCAG 2.1 AA compliant)

## Dependencies
- Depends on: #2, #4 (Architecture, Database)

## Labels
`frontend`, `javascript`, `checkout`, `priority:high`, `phase:3`

**Comment by dangoriaynov** [comment]
Implemented in PR #26

The modern JavaScript architecture is now complete and active:

✅ Created 3 modular JS files (1,075 total lines):
  - sesh-checkout.js (469 lines)
  - sesh-location-selector.js (359 lines)
  - sesh-price-display.js (247 lines)

✅ Removed security vulnerabilities:
  - No eval() usage
  - Nonce verification on all AJAX calls
  - Proper input sanitization and output escaping

✅ Eliminated performance issues:
  - Removed 3 polling loops (500ms intervals)
  - Event-driven architecture using WooCommerce's updated_checkout event
  - Cached delivery prices

✅ Improved maintainability:
  - Modular architecture with separation of concerns
  - No global variables
  - Proper dependency management
  - JSDoc comments

The legacy js.php is now disabled via the is_new_system_ready() flag.
